src/python/grpcio/commands.py

```diff
@@ -184,6 +184,70 @@ def run(self):
     build_py.build_py.run(self)
 
 
+def _poison_extensions(extensions, message):
+  """Includes a file that will always fail to compile in all extensions."""
+  poison_filename = os.path.join(PYTHON_STEM, 'poison.c')
+  with open(poison_filename, 'w') as poison:
+    poison.write('#error {}'.format(message))
+  for extension in extensions:
+    extension.sources = [poison_filename]
+
+def check_and_update_cythonization(extensions):
+  """Replace .pyx files with their generated counterparts and return whether or
+     not cythonization still needs to occur."""
+  built = True
+  for extension in extensions:
+    pyx_sources = [source for source in extension.sources if os.path.splitext(source)[1] == '.pyx']
```

Comment:
Changed it to not require list comprehensions, though I do think they were more readable and that performance concerns don't matter here (using `itertools.groupby` increased the line count by five or so and complicated matters with gunky key munging, so didn't go down that route).